### ISSUE TITLE
The value for Number of CPUs must be above zero. You provided 0

### ISSUE THREAD

**Issue by storbel** [vmware/terraform-provider-vcd#875]
Hello,

 Error: error changing cpu size: error updating VM spec section: API Error: 400: [ 5e3b99dd-f095-4622-b5d6-58a0ee5aedb7 ] The value for Number of CPUs must be above zero. You provided 0.
    │
    │   with vcd_vapp_vm.vm-webt-stack[0],
    │   on etlfme.tf line 88, in resource "vcd_vapp_vm" "vm-webt-stack":
    │   88: resource "vcd_vapp_vm" "vm-webt-stack" {

### Terraform Version
Terraform v1.1.6
on linux_amd64

### Affected Resource(s)
Please list the resources as a list, for example:
- vcd_vapp_vm

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files

```hcl
resource "vcd_vapp_vm" "vm-webt-stack" {

  name = format("%s%s%s%s%s%s", "etlfme", "1", "webt", "${count.index + 1}", "n", "0")

  computer_name = format("%s%s%s%s%s%s", "etlfme", "1", "webt", "${count.index + 1}", "n", "0")

  vapp_name     = "lan0etlfme1webt"
  count         = "2"
  catalog_name  = "AppLaunchPad-Catalogue"
  template_name = "ws-2019-dc-gui"


  sizing_policy_id = data.vcd_vm_sizing_policy.c_large_stack.id


  cpu_hot_add_enabled    = true
  memory_hot_add_enabled = true
  power_on               = true
  network {
    name               = "lan_prd_front"
    type               = "org"
    ip_allocation_mode = "POOL"
    is_primary         = true

  }

  metadata = {
    function = "webt"
    env      = "prd"
    zone     = "front"
    region   = "lan"
  }
  lifecycle {
    prevent_destroy = false
  }
  customization {
    force   = false
    enabled = true
    // TODO: Add windows script
    // Update Windows Administrator password
    admin_password                      = "xxxxx"
    allow_local_admin_password          = true
    auto_generate_password              = false
    must_change_password_on_first_login = true
  }


}

```

### Debug Output
Please provide a link to GitHub [Gist(s)](https://gist.github.com) containing complete debug
output. You can enable debug by using the commands below:
```shell
vcd_vapp_vm.vm-webt-stack[0]: Still creating... [13m10s elapsed]
    vcd_vapp_vm.vm-fils-stack[0]: Still creating... [13m10s elapsed]
    vcd_vapp_vm.vm-appl-stack[0]: Still creating... [13m10s elapsed]
    vcd_vapp_vm.vm-webt-stack[0]: Still creating... [13m20s elapsed]
    vcd_vapp_vm.vm-fils-stack[0]: Still creating... [13m20s elapsed]
    vcd_vapp_vm.vm-appl-stack[0]: Still creating... [13m20s elapsed]
    ╷
    │ Error: error changing cpu size: error updating VM spec section: API Error: 400: [ 5e3b99dd-f095-4622-b5d6-58a0ee5aedb7 ] The value for Number of CPUs must be above zero. You provided 0.
    │
    │   with vcd_vapp_vm.vm-webt-stack[0],
    │   on etlfme.tf line 88, in resource "vcd_vapp_vm" "vm-webt-stack":
    │   88: resource "vcd_vapp_vm" "vm-webt-stack" {
    │
    ╵
    ╷
    │ Error: error changing cpu size: error updating VM spec section: API Error: 400: [ 8bbf4b03-e544-4e94-91ed-d1a7a58023a6 ] The value for Number of CPUs must be above zero. You provided 0.
    │
    │   with vcd_vapp_vm.vm-webt-stack[1],
    │   on etlfme.tf line 88, in resource "vcd_vapp_vm" "vm-webt-stack":
    │   88: resource "vcd_vapp_vm" "vm-webt-stack" {
    │
    ╵
    ╷
    │ Error: error changing cpu size: error updating VM spec section: API Error: 400: [ fe2fd24f-fe3d-45eb-9d61-e8cc07d3b010 ] The value for Number of CPUs must be above zero. You provided 0.
    │
    │   with vcd_vapp_vm.vm-appl-stack[0],
    │   on etlfme.tf line 145, in resource "vcd_vapp_vm" "vm-appl-stack":
    │  145: resource "vcd_vapp_vm" "vm-appl-stack" {

```

### Expected Behavior
create the virtual machines without errors

### Actual Behavior
It request the cpu count, but it's already accepting sizing_policy

**Comment by adambarreiro** [comment]
Hi @storbel, thanks for reporting this issue.

I could reproduce it using latest version of the Provider and VCD 10.3.
I'll check further.

**Comment by adambarreiro** [comment]
A fix is included in [v3.7.0](https://github.com/vmware/terraform-provider-vcd/releases/tag/v3.7.0).
Thanks!
